core/src/autogluon/core/constants.py

```diff
@@ -25,6 +25,12 @@
 AUTO_WEIGHT = 'auto_weight'
 BALANCE_WEIGHT = 'balance_weight'
 
+# Constants used to infer problem type
+MULTICLASS_LIMIT = 1000  # assume regression if dtype is numeric and unique label count is above this limit
```

Comment:
[minor] probably name it more explicitly to tell if it is an upper_bound limit or lower_bound limit